component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAuthenticator.java

```diff
@@ -553,42 +551,33 @@ private boolean isValidTokenLocalUser(String code, String username, Authenticati
             if (userRealm != null) {
                 Map<String, String> userClaimValues = userRealm.getUserStoreManager()
                         .getUserClaimValues(tenantAwareUsername, new String[]{BACKUP_CODES_CLAIM}, null);
-                String encryptedBackupCodes = userClaimValues.get(BACKUP_CODES_CLAIM);
-                return isValidBackupCode(code, context, username, encryptedBackupCodes);
+                String hashedBackupCodes = userClaimValues.get(BACKUP_CODES_CLAIM);
+                return isValidBackupCode(code, context, username, hashedBackupCodes);
             } else {
                 throw new BackupCodeException(ERROR_CODE_ERROR_FIND_USER_REALM.getCode(),
                         String.format(ERROR_CODE_ERROR_FIND_USER_REALM.getMessage(),
                                 CarbonContext.getThreadLocalCarbonContext().getTenantDomain()));
             }
-        } catch (UserStoreException e) {
+        } catch (UserStoreException | NoSuchAlgorithmException e) {
             throw new BackupCodeException(ERROR_CODE_ERROR_ACCESS_USER_REALM.getCode(),
                     String.format(ERROR_CODE_ERROR_ACCESS_USER_REALM.getMessage(), tenantAwareUsername, e));
         }
     }
 
     private boolean isValidBackupCode(String token, AuthenticationContext context, String userName,
-                                      String encryptedBackupCodes) throws BackupCodeException {
+                                      String hashedBackupCodes) throws BackupCodeException, NoSuchAlgorithmException {
 
-        if (StringUtils.isBlank(encryptedBackupCodes)) {
+        if (StringUtils.isBlank(hashedBackupCodes)) {
             if (log.isDebugEnabled()) {
                 log.debug("No backup codes found for user: " + userName);
             }
             return false;
         }
-        String decryptedCodes = null;
-        try {
-            if (StringUtils.isNotEmpty(encryptedBackupCodes)) {
-                decryptedCodes = BackupCodeUtil.decrypt(encryptedBackupCodes);
-            }
-        } catch (CryptoException e) {
-            throw new BackupCodeException(ERROR_CODE_ERROR_DECRYPT_BACKUP_CODE.getCode(),
-                    ERROR_CODE_ERROR_DECRYPT_BACKUP_CODE.getMessage(), e);
-        }
         List<String> backupCodeList;
-        if (StringUtils.isEmpty(decryptedCodes)) {
+        if (StringUtils.isEmpty(hashedBackupCodes)) {
```

Comment:
Why do we need this? We already have a check right?

Reply:
addressed with [7d9674b](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/6/commits/7d9674b6bf172b43c30479b0d64fdfa6f8130373)